### PR TITLE
Downgrade `sphinx-autodoc-typehints`

### PR DESCRIPTION
**Pull request by qgallouedec**
## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) for new features and bug fixes)

closes https://github.com/DLR-RM/stable-baselines3/issues/1290

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist

- [x] I've read the [CONTRIBUTION](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) guide (**required**)
- [ ] I have updated the changelog accordingly (**required**).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.
- [x] I have opened an associated PR on the [SB3-Contrib repository](https://github.com/Stable-Baselines-Team/stable-baselines3-contrib) (if necessary)
- [x] I have opened an associated PR on the [RL-Zoo3 repository](https://github.com/DLR-RM/rl-baselines3-zoo) (if necessary)
- [x] I have reformatted the code using `make format` (**required**)
- [x] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [x] I have ensured `make pytest` and `make type` both pass. (**required**)
- [x] I have checked that the documentation builds using `make doc` (**required**)

Note: You can run most of the checks using `make commit-checks`.

Note: we are using a maximum length of 127 characters per line

### CODE REVIEW

**Comment by qgallouedec** [comment]
I don't know why it would break pytype now... Whatever, I'm hot-fixing this...

File: setup.py [hunk 1]
```diff
@@ -117,7 +117,7 @@
             # For spelling
             "sphinxcontrib.spelling",
             # Type hints support
-            "sphinx-autodoc-typehints",
+            "sphinx-autodoc-typehints==1.21.1",  # TODO: remove version constraint, see #1290
```
Comment (araffin) [inline setup.py:120]: after a quick check, gym 0.26 doesn't solve that issue... so I guess best is to override the docstrings (as long as there are not too many)

File: stable_baselines3/common/buffers.py [hunk 2]
```diff
@@ -474,7 +474,7 @@ def get(self, batch_size: Optional[int] = None) -> Generator[RolloutBufferSample
             yield self._get_samples(indices[start_idx : start_idx + batch_size])
             start_idx += batch_size
 
-    def _get_samples(self, batch_inds: np.ndarray, env: Optional[VecNormalize] = None) -> RolloutBufferSamples:
+    def _get_samples(self, batch_inds: np.ndarray, env: Optional[VecNormalize] = None) -> RolloutBufferSamples:  # type: ignore[signature-mismatch] #FIXME
```
Comment (araffin) [inline stable_baselines3/common/buffers.py:477]: probably due to a newer pytype version

File: setup.py (hunk 1 above)
Comment (qgallouedec) [reply setup.py:120]: Wonder if it's not simply a bug in `sphinx-autodoc-typehints`.

Comment (araffin) [reply setup.py:120]: looks like so

**Review by araffin — APPROVED** [review]
LGTM, thanks =)

**Pull request merged**